FWCore/Catalog/src/FileLocator.cc

```diff
@@ -112,19 +115,19 @@ namespace edm {
     // a `getElementsByTagName()` in the calling method.
     DOMElement* ruleElement = static_cast<DOMElement *>(ruleNode);
 
-    std::string const protocol = _toString(ruleElement->getAttribute(_toDOMS("protocol")));
-    std::string destinationMatchRegexp = _toString(ruleElement->getAttribute(_toDOMS("destination-match")));
+    std::string const protocol = _toString(ruleElement->getAttribute(m_str_protocol));
```

Comment:
This would be
```C++
std::string const protocol = _toString(ruleElement->getAttribute(OwningXMLStringPtr{XMLString::transcode("protocol")}.get()));
```

To simplify further you could bring back `_toDom` which just creates the `OwningXMLStringPtr` by calling `XMLString::transcode`. Then these lines stay exactly the way the were originally.